### PR TITLE
Bump Herwig to v7.2.0, ThePEG to v2.2.0

### PR DESCRIPTION
**Pull request by mfasDa**
- Bump Herwig to v7.2.0
- Bump ThePEG to v2.2.0 (dependency of Herwig)
- Add CT14lo and CT14nlo to lhapdf-pdfsets,
  required by Herwig v7.2.0

### CODE REVIEW

File: herwig.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
 package: Herwig
 version: "%(tag_basename)s"
-tag: "v7.1.2-alice1"
-source: https://github.com/alisw/herwig
+tag: "alice/v7.2.0"
+source: https://github.com/mfasDa/herwig
```
Comment (qgp) [inline herwig.sh]: Ok for testing, but for the final deployment this should be changed to a tag (not a branch) from alisw.

File: thepeg.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package: ThePEG
 version: "%(tag_basename)s"
-tag: "v2.1.2"
-source: https://github.com/alisw/thepeg
+tag: "alice/v2.2.0"
+source: https://github.com/mfasDa/thepeg
```
Comment (qgp) [inline thepeg.sh]: Ok for testing, but for the final deployment this should be changed to a tag (not a branch) from alisw.

File: herwig.sh (hunk 1 above)
Comment (mfasDa) [reply herwig.sh]: Of course, preparing the branch for Herwig 7.2.

File: thepeg.sh (hunk 2 above)
Comment (mfasDa) [reply thepeg.sh]: Of course, preparing the branch for ThePEG 2.2.

File: herwig.sh (hunk 1 above)
Comment (qgp) [reply herwig.sh]: @mfasDa merged and tagged as v7.2.0

File: thepeg.sh (hunk 2 above)
Comment (qgp) [reply thepeg.sh]: @mfasDa merged and tagged as v2.2.0

**Comment by mfasDa** [comment]
Checking whether adding GMP as build requirement for ThePEG fixes the compiler issue of Herwig.

**Comment by marcovanleeuwen** [comment]
Hi Markus, @ktf,

It looks like the test-compilation above is not using defaults-generators. It is using fastjet v3.2.1 instead of v3.3.3. The -lgmp issue here is likely caused by that.

(fixing it here will not fix it for running on the grid yet; for that we need to fix the relocation issue)

Marco.

**Comment by qgp** [comment]
@ktf Fine with me, but has the problem of using the wrong defaults (pointed out by Marco in the previous comment https://github.com/alisw/alidist/pull/2153#issuecomment-605009067) been addressed?

**Comment by mfasDa** [comment]
Could one trigger a new build test?

**Comment by mfasDa** [comment]
@maireiphc @preghenella I forced triggering the build test again and now it picked up the proper defaults, and all tests passed. I think one could proceed now.

**Comment by mfasDa** [comment]
@qgp The build test is passing now - can the PR be approved now?

**Comment by preghenella** [comment]
hi @mfasDa ,
we have a issue with building `AliGenerators` in these days, that is failing in `ThePEG`.
I fear that if we bring this PR in, it might be harder to understand what is wrong.

Well, it would even be that this fixes it, but given that it appeared without any relation to a `ThePEG` upgrade, I would tend to wait a bit.

**Comment by preghenella** [comment]
@mfasDa , we got a successful build of `AliGenerators` this morning.
Thanks to @ktf for helping with that.

I think this PR can be merged.

**Comment by mfasDa** [comment]
Thanks a lot!

Once the PR is merged may I propose to think about decommissioning couple of generators which are shipped inside AliRoot in favour of linking against packages from AliGenerators. This affects PYTHIA(6,8), and LHAPDF. For the Herwig version (5) in AliRoot I am not sure it is currently in use but people should rather switch to Herwig7.

**Comment by preghenella** [comment]
Ciao @mfasDa ,
it would be possibly a good thing to do, but I am not sure how easy this is (will probably need to touch cmake in several places), who is supposed to decide on that step (Offline Board I guess) and who will actually execute the task...

**Comment by preghenella** [comment]
ciao @ktf , can we merge this PR?

I fear that we might encounter again the problem with YODA while building on Jenkins, given that we are upgrading ThePEG.
In case, we know how to deal with it... even though it is not an elegant way (we'll think about the right way to do it if indeed we hit the problem again).

Thanks,
Roberto

**Comment by preghenella** [comment]
that was perhaps expected.
@mfasDa , `AliGenerators` fails in building `ThePEG`.
https://alijenkins.cern.ch/job/DailyBuilds/job/DailyAliGenerators/354/console

The problems is that is seems that `ThePEG` does not find `YODA`, unless the `YODA` package is built in the same session.
I would say that there is a good reason to think that there is a issue in how `ThePEG` is pointing at `YODA`.
This conclusion comes from the fact that a recent PR in `Rivet` (that also uses `YODA`) did not show problems related to `YODA`.
So, `Rivet` is able to find `YODA` properly, `ThePEG` is not.

Do you have an idea what could be the problem?
I did not manage to reproduce it locally, this must have something to do with relocation matters.

We know that if we trigger a rebuild of `YODA` we can solve it.
But the problem will re-appear in the future, so I would be happy to find a solution forever.

Thanks,
Roberto

ps: @ktf , I know this is a hack, but is there a mechanism via `alidist` receipts that can be used to force a package build from another package? I mean, could `ThePEG` signal that he wants a `YODA` rebuild?

**Comment by ktf** [comment]
just add an extra space to yoda...

**Comment by ktf** [comment]
or a comment...

**Comment by preghenella** [comment]
uhm... maybe it is `Rivet` the problem and `ThePEG` relies on something that `Rivet` provides to locate `YODA`.
Will have a look

**Comment by preghenella** [comment]
what I meant @ktf was if one can do something in the receipt of `ThePEG` to force a rebuild on `YODA`.
Instead of the manual trigger via `alidist` update

**Comment by preghenella** [comment]
nevermind, for some reason today I can reproduce the error on my computer.
I think because I have downloaded `YODA` from `alicache`. let's see.

**Comment by ktf** [comment]
I think the correct thing to do is to understand what is happening, as it's effectively a bug. I suspect the problem is that we pick up YODA from rivet, rather than directly.

**Comment by preghenella** [comment]
I'm on it.
Somewhere while building `ThePEG` something is doing this `grep` and this `sed` 
```
/usr/bin/grep: /mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/YODA/v1.8.2-2/lib/libYODA.la: No such file or directory
/usr/bin/sed: can't read /mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/YODA/v1.8.2-2/lib/libYODA.la: No such file or directory
libtool:   error: '/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/YODA/v1.8.2-2/lib/libYODA.la' is not a valid libtool archive
```
but I cannot tell where.

Looks like explicitly adding `YODA` to the requirements of `ThePEG` does the trick.

Will push a commit and open PR elsewhere, let's see it is works.

**Comment by ktf** [comment]
Try grepping for YODA in Rivet. I think it caches some YODA path which gets out of sync.

**Comment by preghenella** [comment]
ps: I think `ThePEG` is able to correctly pick the correct `YODA` in general in this build.
In the `Jenkins` console you can find several instances where is works ok

```
unknown-linux-gnu/7.3.0/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/pythia/v8243-alice1a-1/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/Rivet/3.1.1-alice1-1/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/fastjet/v3.3.3_1.042-alice1-24/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/cgal/4.6.3-34/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/MPFR/v3.1.3-63/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/GMP/v6.0.0-58/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156153/1/slc7_x86-64/YODA/v1.8.2-2/lib -
```

see here, `YODA` is taken from build `20156153`.

The line of the failure picks from build `20156152`.

**Comment by preghenella** [comment]
I think I have found the reason for the failure, although I don't know how to handle the problem yet.

Looks like at some point `ThePEG` is using a `libtool` file from `Rivet`.
```
$RIVET_ROOT/lib/libRivet.la
```
I am not familiar with `libtool` but my feeling is that it looks into this file to determine the location of some dependencies.
And unfortunately, the file contains information about the absolute paths at the `Rivet` build.
```
dependency_libs=' -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/YODA/v1.8.2-2/lib -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/HepMC/HEPMC_02_06_10-5/lib -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/MPFR/v3.1.3-63/lib -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/GMP/v6.0.0-58/lib -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/cgal/4.6.3-34/lib -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/boost/v1.72.0-alice1-37/lib -L/usr/lib -lboost_thread -lboost_system /mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/YODA/v1.8.2-2/lib/libYODA.la -ldl -L/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/fastjet/v3.3.3_1.042-alice1-24/lib -lfastjet -lCGAL -lgmp -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156144/1/slc7_x86-64/cgal/4.6.3-34/lib -L/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156144/1/slc7_x86-64/GMP/v6.0.0-58/lib -lm -lfastjetplugins -lsiscone_spherical -lsiscone -lfastjetcontribfragile -lfastjettools -lHepMC -lz'
``` 
If I manually edit the the line
```
/mnt/mesos/sandbox/sandbox/jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156152/1/slc7_x86-64/YODA/v1.8.2-2/lib/libYODA.la
```
which is the line where `ThePEG` complains about, with the correct path on my system, I can make `ThePEG` compile.

I will have to find out how things can be properly done when relocating.
If you have hints where to put hands, thanks and I will have a look.

Roberto

**Pull request merged**